Service/src/routes/league.py

```diff
@@ -1,38 +1,43 @@
 from flask import Blueprint, request, Response, g, jsonify, make_response
 import simplejson as json
-import time
+from datetime import datetime
 
+from auth import auth
 import routes.portfolio as portfolio
+from request_validator import validator
+from request_validator.schemas import league_post_schema
 from util.utility import Utility
 from util.error_map import errors
 
 league_api = Blueprint('league_api', __name__)
 
 DATE_FORMAT = "%m-%d-%Y"
+DAYS_IN_YEAR = 365
 
-
-@league_api.route('/api/league', methods=['POST'])
+@league_api.route('/api/v1.0/leagues', methods=['POST'])
+@auth.login_required
+@validator.validate_body(league_post_schema.fields)
 def post_league():
    body = request.get_json()
-   try:
-      result = LeagueSchema().load(body)
-   except ValidationError as err:
-      return make_response(json.dumps(err.messages), 400)
-
+   startDate = datetime.strptime(body['start'], DATE_FORMAT)
+   endDate = datetime.strptime(body['end'], DATE_FORMAT)
+   if startDate > endDate:
+      return make_response(jsonify(EndBeforeStart=errors['endBeforeStart']), 400)
+   
+   leagueDuration = (endDate - startDate).days
+   print(str(leagueDuration))
+   if leagueDuration > DAYS_IN_YEAR:
+      return make_response(jsonify(LeagueDurationTooLong=errors['leagueDurationTooLong']), 400)
+
+   startPos = body.get('startPos') or 10000
```

Comment:
Do we need this default value? I feel like the start position should always have a value given by the user if they're starting a league.

Reply:
I like having defaults whenever possible...if the user 'should always' give the start pos, then that's fine. Just means the default value won't be used. Im gonna get rid of this magic number tho haha.